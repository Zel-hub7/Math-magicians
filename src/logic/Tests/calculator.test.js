import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../../Components/calculator';

test('performs addition', () => {
  const { getByText, queryByText } = render(<Calculator />);

  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('='));

  const display = queryByText('3');
  expect(display).toBeTruthy();
});

test('matches snapshot', () => {
  const { container } = render(<Calculator />);
  expect(container).toMatchSnapshot();
});
