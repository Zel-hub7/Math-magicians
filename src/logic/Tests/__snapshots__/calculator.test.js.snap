// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="main-container"
  >
    <div
      class="desc"
    >
      <h1
        class="title"
      >
        LET'S DO SOME MATH
      </h1>
    </div>
    <div
      class="calculator"
    >
      <input
        class="calculator-screen"
        disabled=""
        type="text"
        value="0"
      />
      <div
        class="calculator-grid"
      >
        <button
          type="button"
        >
          AC
        </button>
        <button
          type="button"
        >
          +/-
        </button>
        <button
          type="button"
        >
          %
        </button>
        <button
          class="oper"
          type="button"
        >
          รท
        </button>
        <button
          type="button"
        >
          7
        </button>
        <button
          type="button"
        >
          8
        </button>
        <button
          type="button"
        >
          9
        </button>
        <button
          class="oper"
          type="button"
        >
          x
        </button>
        <button
          type="button"
        >
          4
        </button>
        <button
          type="button"
        >
          5
        </button>
        <button
          type="button"
        >
          6
        </button>
        <button
          class="oper"
          type="button"
        >
          -
        </button>
        <button
          type="button"
        >
          1
        </button>
        <button
          type="button"
        >
          2
        </button>
        <button
          type="button"
        >
          3
        </button>
        <button
          class="oper"
          type="button"
        >
          +
        </button>
        <button
          class="span-two"
          type="button"
        >
          0
        </button>
        <button
          type="button"
        >
          .
        </button>
        <button
          class="oper"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
